Review of Milestone 2.

cod3smith opened the pull request:
This PR herein contains the following:
1. A way for a user to add friends.
2. Unit tests for all models.
3. Integration tests for the user view.

cod3smith committed:
Setting up project.

Commit:
add gems

Commit:
add friendship model and edit user model

Commit:
add foreign key

cod3smith committed:
Add way to make friends

Commit:
start to add testing

cod3smith committed:
Write unit and integration tests.

cod3smith committed:
Merge branch 'development' into milestone2

trekab reviewed (changes requested):
### Changes Required! :recycle: 
Hi team! Well done implementing the required relationships amongst your models. There are some requirements that you still need to implement to complete this milestone though. These include:
- [x] A user should be able to see friendship requests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] A user has to accept friendship request to start friendship ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] A user should be able to reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] The button to invite a user to friendship should appear on both the users' list page and on a single user's page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] Posts timeline should include posts written by the logged-in user and his/her friends. Your current version displays all posts to any logged-in user. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] Well done writing the tests for your associations. Consider adding some validation tests and integration tests. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [x] __[Note]__ It appears that you set up your linting correctly but it is still unable to run the checks on your pull request. Consider replacing your config files with new ones and see if this persists.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

cod3smith committed:
applications as a social media application
friends controller to enable rejecting and accepting of requests.
This is a complete feature for viewing friend requests.

Commit:
Create Friendship page

Commit:
issues created

cod3smith committed:
Add the custom timeline feature.

cod3smith committed:
Improve navbar.

Commit:
Merge branch 'milestone2'

cod3smith committed:
Reverse order of the posts to show latest posts first.

cod3smith committed:
Merge remote-tracking branch 'origin/milestone2' into milestone2

cod3smith committed:
Add .idea to .gitignore

cod3smith committed:
Add Github Actions.

cod3smith committed:
Add Github Actions.

cod3smith committed:
Deploy project to Heroku.

cod3smith committed:
Fixing deployment on Heroku.

cod3smith committed:
Fix rubocop offenses.

Commit:
fix unfriend issues

cod3smith committed:
Fix the user show page.

cod3smith committed:
Change status of issue in issue.txt.

Commit:
fix user index request issue

Commit:
Merge branch 'milestone2' of github.com:abouhid/Social_media_app into milestone2

Commit:
Update README.md

cod3smith committed:
Add gif to the README.md

cod3smith committed:
Add gif to the README.md

cod3smith committed:
Fix readme

Commit:
add test and issues

Commit:
Merge branch 'milestone2' of github.com:abouhid/Social_media_app into milestone2

cod3smith committed:
Add tests for friendships and users.

Commit:
add Log out test

cod3smith committed:
Add more tests on friendship to test more cases.

cod3smith committed:
Cleaning up the tests.

cod3smith committed:
Fix merge conflict.

Commit:
finished tests for posts

Commit:
finished tests for posts

Commit:
apply rubocop

cod3smith committed:
Add procedures on how to run tests to readme.

abouhid commented:
All of the issues were fixed and we even took it further and finished the following two milestones! Thank you for the help.

Commit:
Delete Issues.txt

cod3smith committed:
Add procedures on how to run tests to readme.

cod3smith committed:
Merge branch 'milestone2' of github.com:abouhid/Social_media_app into milestone2

kubilaycaglayan commented on app/views/users/show.html.erb:
```diff
@@ -1,6 +1,19 @@
+
+    <article class="timeline">
+     
+    <% if @user.friendship_requested?(current_user) %>
+
+    <ul class="posts">
+      <%= render "friend_request" %>
+    </ul>
+    <% end %> 
+
+ <% unless current_user.friendship_requested?(@user) || current_user.friend?(@user) %>
```
:bulb: There should be no logic in the views.

kubilaycaglayan reviewed (changes requested):
Hello Team 👋
Status: Changes Required ♻️

Great job! You fulfilled the requirements that the previous TSE mentioned. 👍👏😊
There's just one thing I noticed when looking at show.html.erb in users view.
We should keep the logic away from the views. 
Please work on it, I am sharing the necessary documentation with you.


### - [ ] [REQUIRED] There shoud be absolutely NO logic in views.

>Views are only for displaying information that became available to it from its corresponding controller not generating it.

**Resources** 
- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

### That's it, team! After making the changes you can request another code review 😇

With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

cod3smith committed:
Implement TSE reviews.

cod3smith committed:
Add user helper.

Commit:
change colors

Commit:
Merge branch 'milestone2' of github.com:abouhid/Social_media_app into milestone2

kobiyoyo reviewed (approved):
# Project Approved :thumbsup:
## Hello Team  ,  you have done a brilliant job with this project. :clap:
#### You can merge !!!

voscarmv reviewed (approved):
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️
